### ISSUE TITLE
cluster tag doesn't allow more than 8 freeform tags.

### ISSUE THREAD

**Issue by shb-mll** [oracle-terraform-modules/terraform-oci-oke#698]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description 
The module allows only 8 **cluster** freeform tags compared to 9 freeform tags for persistent volume and  loadbalancer tags.
Adding the 9th tag fail's  the module with below error message. 
![image](https://user-images.githubusercontent.com/128678535/231399669-8b93a05a-3e95-42a9-9adf-213860cc1889.png)

could you check and update on how to set 9 freeform tag for cluster. (This works when tags are added using oracle cloud UI).

### Terraform Version and Provider Version
$ terraform -v
Terraform v1.3.9
on darwin_arm64
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/oracle/oci v4.115.0

### Affected Resource(s)
oci_containerengine_cluster

### Terraform Configuration Files
https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/modules/oke/cluster.tf

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. 
# Please remove any sensitive information from configuration files before sharing them. 
```

### Steps to Reproduce

1. `terraform apply`
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/128678535/231397831-3a861282-7998-4716-bee2-8f6b3cd08711.png">

**Comment by hyder** [comment]
Hi, I'm unable to replicate this issue. I've successfully created a cluster with 9 tags: 

```
  oke = {
    cluster = {
      environment = "dev"
      role        = "cluster"
      3 = "3"
      4 = "4"
      5 = "5"
      6 = "6"
      7 = "7"
      8 = "8"
      9 = "9"
    }
    persistent_volume = {
      environment = "dev"
      2 = "2"
      3 = "3"
      4 = "4"
      5 = "5"
      6 = "6"
      7 = "7"
      8 = "8"
      9 = "9"      
    }
    service_lb = {
      environment = "dev"
      role        = "load balancer"
      3 = "3"
      4 = "4"
      5 = "5"
      6 = "6"
      7 = "7"
      8 = "8"
      9 = "9"      
    }
    node_pool = {
      role = "workers_np"
    }
    node = {
      role = "workers"
    }  
```

**Comment by shb-mll** [comment]
Hi @hyder 
I am  using the [oke](https://github.com/oracle-terraform-modules/terraform-oci-oke/tree/main/modules/oke) module in the modules folder to create the cluster. I tested with your tags and they worked for me too, so it appears the issue is not with the 9th tag but with the tags used. 


The following example tags fail for me, could you test with these and confirm 
```
cluster = {
      Team           = "Developement"
      Service        = "Oracle service"
      Component      = "Kubernetes Cluster"
      Environment    = "dev"
      Cluster        = "dev"
      CostCenter     = "Demo_hybrid_team"
      systemsowner   = "Management Team"
      engineer_owner = "Engineering Team"
      Managed_by     = "Production Deployment"
    }
```

**Comment by hyder** [comment]
Hi,

Can you please elaborate what you mean by "the oke module in the modules folder"? Are you running Terraform from there?

**Comment by shb-mll** [comment]
Hi @hyder 
In my use case I want to create OKE cluster only so I am directly using the OKE module in the modules folder. 
so in my config the source is oke subfolder under modules folder.  The other resources (VCN , subnets , bastion,storage) are created in separate modules.
  
source = "github.com/oracle-terraform-modules/terraform-oci-oke.git//modules/oke?ref=v4.5.7"

**Comment by hyder** [comment]
@devoncrouse, can you please see if the 5.x branch is ready for what @shb-mll wants to do?

**Comment by devoncrouse** [comment]
@hyder @shb-mll You can use something like this with 5.x for existing resources:
```
module "oke" {
  source         = "github.com/oracle-terraform-modules/terraform-oci-oke.git?ref=5.x&depth=1"
  providers      = { oci.home = oci.home }
  tenancy_ocid   = local.tenancy_ocid
  compartment_id = "ocid1.compartment..."

  create_nsgs           = false
  create_bastion        = false
  create_operator       = false
  create_iam_resources  = false
  create_vcn            = false
  assign_dns            = false
  vcn_id                = "ocid1.vcn..."
  ig_route_table_id     = "ocid1.routetable..."
  nat_route_table_id    = "ocid1.routetable..."
  control_plane_nsg_ids = ["ocid1.networksecuritygroup..."]
  worker_nsg_ids        = ["ocid1.networksecuritygroup..."]
  bastion_public_ip     = "xxx.xxx.xxx.xxx"
  ssh_public_key        = "ssh-rsa AAAA...== user@domain"
  subnets = {
    cp      = { id = "ocid1.subnet..." }
    workers = { id = "ocid1.subnet..." }
    int_lb  = { id = "ocid1.subnet..." }
  }

  control_plane_is_public = false
  load_balancers          = "internal"
  preferred_load_balancer = "internal"

  worker_pools = { # see docs/examples for more
    np0 = { size = 1 }
  }
}
```

**Comment by shb-mll** [comment]
Hi @devoncrouse  thanks for the sample code. I will check this branch to see how it fits in my use-case.

**Comment by hyder** [comment]
hi @shb-mll.

Were you able to try the solution from @devoncrouse ?

**Comment by shb-mll** [comment]
Hi @hyder  

Sorry for the delay, I wasn't able to test the 5.x branch however I did test the latest version v4.5.9 and the issue with the tag is solved. I see that the type of freeform_tags is updated to map(map(string)).

**Comment by hyder** [comment]
Ok thanks for the update

**Comment by hyder** [comment]
Resolved in #711
